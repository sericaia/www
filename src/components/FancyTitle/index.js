import React from 'react'
import PropTypes from 'prop-types'

const colors = {
  blue: '#3369E8',
  green: '#009925',
  grey: '#666666',
  yellow: '#EEB211',
  red: '#D50F25',
  pink: '#FB4485',
}

const FancyTitle = ({ title, color = 'blue', lettersNumber = 3 }) => {
  const firstTextToken = title.substring(0, lettersNumber)
  const lastTextToken = title.substring(lettersNumber)

  return (
    <>
      <span style={{ color: colors[color] }}>{firstTextToken}</span>
      {lastTextToken}
    </>
  )
}

FancyTitle.propTypes = {
  title: PropTypes.string.isRequired,
  color: PropTypes.oneOf(Object.keys(colors)),
  lettersNumber: PropTypes.number,
}

export default FancyTitle
