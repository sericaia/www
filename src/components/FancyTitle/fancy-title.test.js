import React from 'react'
import { render } from '@testing-library/react'
import FancyTitle from '.'

describe('FancyTitle', () => {
  it('renders correctly', () => {
    const props = {
      title: 'photography',
      color: 'blue',
      lettersNumber: 5,
    }
    const { container } = render(<FancyTitle {...props} />)
    expect(container).toMatchSnapshot()
  })
})
