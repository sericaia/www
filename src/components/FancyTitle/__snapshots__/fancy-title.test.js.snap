// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FancyTitle renders correctly 1`] = `
<div>
  <h2
    aria-label="photography"
  >
    <span
      style="color: rgb(51, 105, 232);"
    >
      photo
    </span>
    graphy
  </h2>
</div>
`;
